controllers/accounts.js

```diff
@@ -25,6 +25,12 @@ module.exports = {
           date: { $gte: today }
         }).sort({ date: 1 });
 
+        shiftData.forEach(s => {
+          let diff = new Date("1970-1-1 " + s.end_time) - new Date("1970-1-1 " + s.start_time )

```

Comment:
I just checked your `models/Shift` file. Looks like you're storing the start and end times as strings, and the `date` as a Date. I wouldn't recommend this approach. I'd recommend having `start_at` and `end_at` or similar, which should *both* be Date fields.

Consider the scenario of a cafe that stays open past midnight. You might have a shift that runs from 8PM-3AM. What is the `date` of this shift? What would be the start and end times? I think this code you added here would break in that scenario.

That might be a significant refactor, as I assume that a lot of your features probably rely on these times. You don't necessarily need to change it now if you don't want to invest the time, but I'd definitely recommend against storing times as strings ever in the future: it's not very resilient.

I'm not sure what to recommend for this particular piece of code. It definitely not going to work in the scenario I described. It also has a ton of "magic numbers". We tend to use that term for unexplained numbers appearing in code. This seems like it could be hard to maintain. If you want to go ahead with a refactor to stop storing these times as strings, I'd be happy to review such a PR.

If nothing else, I think you should move this logic out of the controller and onto the `Shift` model, maybe as a computed attribute (ie `get length() { // your code here`). This logic is directly related to the way that you're storing data in the DB, I don't think it belongs in the controller.